CONTRIBUTING.md

```
@@ -202,7 +202,18 @@ This should almost always be done using the [`xrpl-codec-gen`](https://github.co
 - Merge your changes.
 
 ### Release
-
+1. Please increment the version in `pyproject.toml` and update the `CHANGELOG.md` file appropriately. We follow [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+2. Please select a commit that is suitable for release and create a tag. The following commands can be helpful:
+`git tag -s -a <tag-title> -m "Optional Message describing the tag"`
+`git tag` -- This command displays all the tags in the repository.
+`git push <remote_name, e.g. upstream> tag <tag_title>`
+3. A [Github Workflow](.github/workflows/publish_to_pypi.yml) completes the rest of the Release steps (building the project, generating a .whl and tarball, publishing on the PyPI platform). The workflow uses OpenID Connect's temporary keys to obtain the necessary PyPI authorization.
+As a prerequisite, the PyPI `xrpl-py` project needs to authorize Github Actions as a "Trusted Publisher". This page contains helpful resources: https://packaging.python.org/en/latest/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/#configuring-trusted-publishing
```

Markdownlint

Bare URL used

(MD034, no-bare-urls)